Xplat/src/main/java/vazkii/botania/client/gui/box/BaubleBoxContainer.java

```diff
@@ -121,4 +122,9 @@ public ItemStack quickMoveStack(Player player, int slotIndex) {
 		return itemstack;
 	}
 
+	@Override
+	public void removed(@NotNull Player player) {
+		ItemNBTHelper.setBoolean(box, BaubleBoxItem.TAG_OPEN, false);
```

Comment:
from briefly skimming vanilla, I think you need to check for `!player.level.isClient` here as this method is called on both sides.